opm/material/fluidsystems/blackoilpvt/LiveOilPvt.hpp

```diff
@@ -59,20 +58,20 @@ class LiveOilPvt
     /*!
      * \brief Initialize the oil parameters via the data specified by the PVTO ECL keyword.
      */
-    void initFromDeck(const Deck& deck, const EclipseState& eclState)
+    void initFromDeck(const EclipseState& eclState)
     {
         const auto& pvtoTables = eclState.getTableManager().getPvtoTables();
-        const auto& densityKeyword = deck.getKeyword("DENSITY");
+        const auto& densityKeyword = eclState.getTableManager().getDensityTable();
 
         assert(pvtoTables.size() == densityKeyword.size());
 
         size_t numRegions = pvtoTables.size();
         setNumRegions(numRegions);
 
         for (unsigned regionIdx = 0; regionIdx < numRegions; ++ regionIdx) {
-            Scalar rhoRefO = densityKeyword.getRecord(regionIdx).getItem("OIL").getSIDouble(0);
-            Scalar rhoRefG = densityKeyword.getRecord(regionIdx).getItem("GAS").getSIDouble(0);
-            Scalar rhoRefW = densityKeyword.getRecord(regionIdx).getItem("WATER").getSIDouble(0);
+            Scalar rhoRefO = densityKeyword[regionIdx].oil;
```

Comment:
> it really doesn't have to be.

sure, but that's an implementation detail. in my experience it was often beneficial to have the flexibility of an accessor method because you sometimes want to change the implementation later (e.g., for performance reasons or to fix bugs).